Tools/baselineDef.cc

```diff
@@ -544,6 +735,44 @@ AK8Flag BaselineVessel::FlagAK8FromCSV(Constituent &ak8) const
   return NoTagNob;
 }       // -----  end of function BaselineVessel::FlagAK8FromCSV  -----
 
+// ===  FUNCTION  ============================================================
+//         Name:  BaselineVessel::FlagAK8DeepFromCSV
+//  Description:  
+// ===========================================================================
+AK8Flag BaselineVessel::FlagAK8DeepFromCSV(unsigned int AK8index) const
+{
+  unsigned loosebcnt =0 ;
+  unsigned mediumbcnt = 0;
+
+  const std::vector<std::vector<TLorentzVector> > &subjets = tr->getVec<std::vector<TLorentzVector> >(UseNoLepVar("puppiAK8SubjetLVec"));
+  const std::vector<TLorentzVector> &jets = tr->getVec<TLorentzVector>(jetVecLabel);
+  const std::vector<float> &CSV = tr->getVec<float>(CSVVecLabel);
+
+  for(auto sub : subjets.at(AK8index))
+  {
+    for(unsigned int ij=0; ij<jets.size(); ij++)
+    {
+      if (sub.DeltaR(jets.at(ij)) < 0.4)
+      {
+        if (jets.at(ij).Pt() < 20 || fabs(jets.at(ij).Eta()) > 2.4) continue;
+        if (UseDeepCSV)
```

Comment:
Do we want this tied to the global "UseDeepCSV" flag?  If I am not mistaken we currently don't have deepCSV sub-jet info saved in the tuples, though this is something to add.

Reply:
Here we are not using sub-jet b-tagging. Jets are AK4 jets here. Not sure how efficiency subjet b-tagging is yet.

Reply:
We should perhaps look into this.  We should have all the experts on this topic descending on the LPC next month if we do not find opportunity before.